src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.State.cs

```diff
@@ -237,12 +209,12 @@ private sealed class FinalState : State
                     Compilation compilationWithoutGeneratedFiles,
                     bool hasSuccessfullyLoaded,
                     ImmutableArray<SourceGeneratedDocumentState> generatedDocuments,
-                    UnrootedSymbolSet? unrootedSymbolSet)
+                    ProjectId projectId,
+                    Dictionary<MetadataReference, ProjectId>? metadataReferenceToProjectId)
```

Comment:
May want a comment here that this _isn't_ being captured and held onto. At first I flagged this during the review going "why is this mutable" and since it's not actually being held it's OK...but kinda funky.